[Feature] support desktop version

gotchazipc opened the pull request:
[Feature] support desktop version
- now webida runs on electron
- many features are missing but will be enabed soon.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/webida/webida-client/913)

<!-- Reviewable:end -->

Commit:
[FEATURE] support electron

 - added some work-around code to use same dojo profile with browser,
  renaming CJS require/module to nrequire, nmodule
 - fixed site-config.json for conn server url typo

Commit:
added site-config-desktop.json

 - fixed old site-config*.json files to have right 'build' property

Commit:
change require for webida-*.js and added new server-api skeletons

 - fixed a small bug in webida-0.3.js, mountByFSID(), to use proper fs server url
 - fixed FS Cache to use mountByFSID, not mount()
 - many plugins & common modules  now requires webida-server-api
   aliased to webida-0.3 by default.
 - added support to save browser window postiion & size when closing window
   in desktop version
 - forced to use plugin-settings-desktop.json instead of plugin-settings.json
   in desktop version
 - added some skeleton codes in server-api-0.1
  (now server-api are modulized in server-api-*-lib/)

Commit:
implemented webida-service-api

 - now basic auth/fs service works. can read/write files via generated swagger client
   with a little bit of patch
 - removed some obsolute bower components not used in webida client anymore
 - added .bowerrc to resolve repositories with https, not ssh
 - reordring & some fixing in dojoConfig.js

Commit:
fix jshit errors & misc

 - fixed jshitrc to respect function hoisting
 - fixed some errors from doublequote
 - removed some logs in server-api
 - fixed pubsub init error

Commit:
fix more jshint errors

 - reordered some variable positions

Commit:
yet another jshint fix

 - aligned .jshintrc file with code convention checking script in jenkins server
 - more fixes

Commit:
removed duplicated functions in FSCache

 - fixed invalid undef value in .jshintrc, too.

Commit:
last fixing on FSCache

 - added jshint directives to set latedef = false

hwshim commented on common/src/webida/serer-api-0.1-lib/fs.js:
```diff
@@ -0,0 +1,30 @@
+/**
```
/serer-api-0.1-lib/ -> /server-api-0.1-lib/

hwshim commented on common/src/webida/serer-api-0.1-lib/fs.js:
```diff
@@ -0,0 +1,30 @@
+        mountByFSID : mountFileSystemById,
+        FileSystem : FileSystem
+    };
+});
```
Please add new line

hwshim commented on common/src/webida/server-api-0.1-lib/WfsMount.js:
```diff
@@ -0,0 +1,248 @@
+            var callable = wfsApi[apiName];
+            var wfsPath = this._fromLegacyPath(path);
+            var args = [ this.wfsId, wfsPath ];
+            for (let i = 2; i < arguments.length; i++) {
```
We should support IE10, the syntax 'let' will cause a problem on IE10. Right?

hwshim commented on common/src/webida/server-api-0.1-lib/auth.js:
```diff
@@ -0,0 +1,83 @@
+/**
```
Use copyright notice and jsdoc description.

/*
 * Copyright (c) 2012-2016 S-Core Co., Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * @file Auth
 * @since 1.0.0
 * @author your.id@samsung.com
 */

Commit:
implemented access token refreshing logic

 - reflect API changes, including login & issueToken
 - new class, TokenManager has been introduced
 - added APL license for each source file, except generated sources
  (we will move the sources into separated project to use with bower)

 - now, ready to implement pubsub messaging events

Commit:
implemented session handling & file system events

 - fixed some bugs on fs-command plugin's invalid command id
 - fixed some bugs on editor context menu typos.
 - fixed editors/DataSourceHandler to use data source with correct URN, fixing '//' to '/'
 - refactoring : moved some wfs related files to server-api-0.1-lib/wfs
 - refactoring : renamed Stats to WfsStats
 - added 'AbstractSocketClient' class, socket-io client wrapper
 - added session module & SessionSocketClient, handling session connection and reconectting
 - added new plugin, session-event-dispatcher, replacing 'resources' plugin
   (can be used with new server api lib only )
 - added WfsEventGate & WfsEventHolder to handle server's file system event from session client
 - added some properties in server api (webida) interface,
   to use session client & some useful informations

Commit:
extracted restful api source to bower component

 - removed webida-service-api-0.1 dir and added webida-restful-api bower dependency
 - removed uid-menu plugin in plugin-settings-desktop.json
 - fixed some errors due to removing uid-menu-plugin
 - implemented off-line cache service (not applied to fs module yet)
 - removed some too-verbose logs
 - fixed many misc. bugs.

Commit:
replaced server-api to webida-service-client bower component

 - fixed bower.json
 - upgraded lodash to 4.14.0
 - fixed some codes that use _.contains(), deprecated long before and now removed
 - renamed server-pubsub-0.x to server-pubsub-compat, for the code does nothing actually

Commit:
updated webida-service-client version

Commit:
applied new server event scheme

 - fixed some lodash-related bugs on SimplePage
 - updated service client to 0.8.1
